src/seismometer/data/timeseries.py

```diff
@@ -77,9 +79,9 @@ def _limit_data(
 ) -> pd.DataFrame:
     """Reduces the data to only include valid data points."""
     include = dataframe[reftime].notna()
-    # boolean events - filter out -1 where invalid
+
     if boolean_event:
-        include = include & (dataframe[event_col] >= 0)
+        include = include & (is_valid_event(dataframe, event_col, reftime))
```

Comment:
Just calling out this is the important change, where knowing the event + ref time we can know that we predicted the correct value.